### PR TITLE
[#10] feature: add use case for creating game

### PR DESCRIPTION
**Pull request by metalalive**
Detail
- declare relevant methods to in-memory repository and domain models `Game` and `Player`
- add shared server context that keeps the repository and configuration

Note:
- @teds-lin please help to review the PR, also let me know if you have other ideas about naming the interface of these domain models and repositories.
  - 請協助檢查 PR, 如果你對這些domain model 和 repository 的介面命名有其他想法，請告訴我.
  - apologize for the big commit.
- @wingtkw , I don't send review request to you cuz you're on vacation, if you'd like to learn any part of the code in the PR, leave your comment here (even after this PR is merged) or start another thread in [GitHub Discussion](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu/discussions)

Thanks

### CODE REVIEW

**Review by teds-lin — COMMENTED** [review]
LGTM
`backend/app/domain/game.py`
有一處疑問，在麻煩釋疑，謝謝。

File: backend/app/domain/game.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+class Game:
+    MAX_NUM_PLAYERS = 4
+
+    def __init__(self, difficult_level: Difficulty = Difficulty.INTRODUCTORY):
+        # TODO, declare setter for difficulty level
+        self._difficulty: Difficulty = difficult_level
+        self._players: List[Player] = []
+
+    def add_players(self, players: List[PlayerDto]):
```
[the remark this answers is not in the document]
Comment (metalalive) [reply backend/app/domain/game.py:29]: > 這邊依照語意，輸入應該是player_dtos而非players，是否理解正確？

是的沒錯
好, 輸入參數也可以修改為 `player_dtos`

**Comment by metalalive** [comment]
teds
please take a look again, the argument now is renamed to `player_dtos` , thanks
請再看一下，參數現在已重新命名為 `player_dtos` ，謝謝

**Review by teds-lin — APPROVED** [review]
LGTM

**Pull request merged**